docs/design/persistent-storage.md

```diff
@@ -0,0 +1,235 @@
+# Persistent Stores
+
+This document proposes a model for managing persistent, cluster-scoped storage for applications requiring long lived data.
+
+### tl;dr
+
+`PersistentStore` objects contain a VolumeSource with a lifecycle greater than the pods it is used by.  Many types of VolumeSources can be implemented as volume plugins, with each implementing an interface for use within a controller.  A pod with persistence requirements (specifically pod.spec.volumes.volume.persistenceRequirements) is bound to a matching VolumeSource from a pool of available stores.  PersistentStore objects give that VolumeSource longevity.
+
+A `PersistentStoreController` manages the creation, attachment, and reclamation of PersistentStore objects for one VolumeSource.  Implementations must be pluggable to support many types of controllers for many types of VolumeSources. The controller can advise the scheduler on host selection, keeping volume rules contained within the provider.  All cloud provider credentials are isolated to the store controller's host.  Every store in the cluster requires a controller to manage the pool (even a pool of 1).
+
+`PersistentStoreBinder` watches for pods with persistence requirements and matches the request with what is available in the pool.
```

Comment:
This is a process or an API object?  Is it a separate process like scheduler?  Is there one per volume type (e.g. one for PD and one for NFS)?